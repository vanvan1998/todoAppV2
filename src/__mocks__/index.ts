const todo = {
  email: 'nguyensivan1998+102@gmail.com',
  id: '82zB2At6gnxs0SbDJ8SW',
  notification: true,
  createdAt: '2024-05-10T01:14:38',
  title: '333333',
  detail: '3333333',
  startTime: '01:00:21',
  startDate: '05/10/2024',
  category: 1,
  completed: false,
  userId: 'j3WklyoTrBg4E1GKNz0Gn0az34V2',
  checkList: []
};
export { todo };
