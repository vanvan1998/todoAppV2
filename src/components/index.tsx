export { AddTodoModal } from './AddTodoModal';
export { Search } from './Search';
export { TodoItem } from './TodoItem';
export { Container } from './Container';
export { Header } from './Header';
export { AddTodo } from './addTodo';
export { SortBy } from './SortBy';
export { Input } from './input';
export { Button } from './button';
