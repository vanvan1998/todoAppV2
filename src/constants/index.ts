export const SORT_BY = {
  NEWEST: 'NEWEST',
  OLDEST: 'OLDEST',
  NONE: 'NONE'
};

export const MODE = {
  UPDATE: 'UPDATE',
  ADD: 'ADD'
};

export const DATE_FORMAT = 'MM/DD/YYYY';
export const TIME_FORMAT = 'HH:mm:ss';
