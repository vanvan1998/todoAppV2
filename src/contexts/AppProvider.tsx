import React from 'react';
import { GlobalLoadingProvider } from './GlobalLoadingProvider';
import { AuthProvider } from './AuthProvider';

interface AppProviderProps {
  children: React.ReactNode;
}

export const AppProvider = ({ children }: AppProviderProps) => {
  return (
    <AuthProvider>
      <GlobalLoadingProvider>{children}</GlobalLoadingProvider>
    </AuthProvider>
  );
};
