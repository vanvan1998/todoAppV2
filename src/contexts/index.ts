export * from './AppProvider';
export * from './GlobalLoadingProvider';
export * from './AuthProvider';
