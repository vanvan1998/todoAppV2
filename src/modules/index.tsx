export * from './left-menu';
export * from './login';
export * from './forgot-password';
export * from './update-password';
export * from './update-profile';
export * from './dash-board';
export * from './sign-up';
