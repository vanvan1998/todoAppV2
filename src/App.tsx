import React from 'react';
import { AuthProvider } from './contexts/AuthContext';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import {
  Signup,
  LeftMenu,
  Login,
  PrivateRoute,
  ForgotPassword,
  UpdatePassword,
  UpdateProfile,
  DashBoard
} from './modules';

function App() {
  return (
    <div style={{ minHeight: '100vh', display: 'flex' }}>
      <Router>
        <AuthProvider>
          <Routes>
            <Route
              path='/update-password'
              element={
                <PrivateRoute>
                  <UpdatePassword />
                </PrivateRoute>
              }
            />
            <Route
              path='/update-profile'
              element={
                <PrivateRoute>
                  <UpdateProfile />
                </PrivateRoute>
              }
            />
            <Route path='/signup' element={<Signup />} />
            <Route path='/login' element={<Login />} />
            <Route path='/forgot-password' element={<ForgotPassword />} />
            <Route
              path='/left-menu'
              element={
                <PrivateRoute>
                  <LeftMenu />
                </PrivateRoute>
              }
            />
            <Route
              path='/'
              element={
                <PrivateRoute>
                  <DashBoard />
                </PrivateRoute>
              }
            />
          </Routes>
        </AuthProvider>
      </Router>
    </div>
  );
}

export default App;
