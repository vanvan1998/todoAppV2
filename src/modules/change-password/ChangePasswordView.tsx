'use client';

import React, { useState } from 'react';
import { useAuth } from '../../contexts/AuthContext';
import styled from 'styled-components';
import { styles } from './ChangePassword.styles';
import { ErrorText, Header, PlaceholderTitle, SuccessText, Title, placeholder } from 'src/theme';
import { Button, Input } from 'src/components';
import { useMediaQuery } from 'src/hooks';
import { isEmpty } from 'lodash';

const Container = styled.div<{ isMobile: boolean }>`
  ${styles.container}
`;

const HeaderWrapper = styled.div<{ isMobile: boolean }>`
  ${styles.headerWrapper}
`;

const TitleWrapper = styled.div`
  ${styles.titleWrapper}
`;

const ErrorWrapper = styled.div<{ isMobile: boolean }>`
  ${styles.errorWrapper}
`;

export const ChangePasswordView = () => {
  const [password, setPassword] = useState('');
  const [passwordConfirm, setPasswordConfirm] = useState('');
  const { currentUser, updatePassword } = useAuth();
  const [error, setError] = useState('');
  const [success, setSuccess] = useState('');
  const [loading, setLoading] = useState(false);
  const { isMobile } = useMediaQuery();

  const handleSubmit = (e: { preventDefault: () => void }) => {
    e.preventDefault();
    if (password !== passwordConfirm) {
      return setError('Passwords do not match');
    }

    const promises = [];
    setLoading(true);
    setError('');
    setSuccess('');
    if (password) {
      promises.push(updatePassword(password) as never);
    }

    Promise.all(promises)
      .then(() => {
        setSuccess('Password updated successfully');
      })
      .catch(() => {
        setError('Failed to update account');
      })
      .finally(() => {
        setLoading(false);
      });
  };

  return (
    <>
      <Container isMobile={isMobile}>
        <HeaderWrapper isMobile={isMobile}>
          <TitleWrapper>
            <Header style={{ fontWeight: '600' }}>Change password</Header>
            <PlaceholderTitle>Please enter your new password to change</PlaceholderTitle>
          </TitleWrapper>
          <Title>
            You are login as <Title style={{ fontWeight: 'bold' }}>{currentUser.email}</Title>
          </Title>
        </HeaderWrapper>
        <Input
          title='New password'
          value={password}
          onChange={setPassword}
          styles={{ marginBottom: 12, marginTop: 40 }}
          inputType='password'
          placeholder='Enter your password...'
          inputStyles={{
            ...(isMobile ? { maxWidth: 320, minWidth: 200, width: '80vw' } : { minHeight: 48, minWidth: 320 }),
            border: `1px solid ${placeholder}`,
            backgroundColor: '#fefeff'
          }}
        />
        <Input
          title='Confirm new password'
          value={passwordConfirm}
          onChange={setPasswordConfirm}
          styles={{ marginBottom: 12 }}
          inputType='password'
          placeholder='Enter your password...'
          inputStyles={{
            ...(isMobile ? { maxWidth: 320, minWidth: 200, width: '80vw' } : { minHeight: 48, minWidth: 320 }),
            border: `1px solid ${placeholder}`,
            backgroundColor: '#fefeff'
          }}
        />
        {success && (
          <ErrorWrapper isMobile={isMobile}>
            <SuccessText>{success}</SuccessText>
          </ErrorWrapper>
        )}
        {error && (
          <ErrorWrapper isMobile={isMobile}>
            <ErrorText>{error}</ErrorText>
          </ErrorWrapper>
        )}
        <Button
          disabled={loading || isEmpty(passwordConfirm) || isEmpty(password)}
          type='submit'
          title='Change Password'
          handleButton={handleSubmit}
          styles={{
            marginTop: 32,
            marginBottom: 16,
            ...(isMobile ? { maxWidth: 320, minWidth: 200, width: '80vw' } : { width: 320 })
          }}
        />
      </Container>
    </>
  );
};
