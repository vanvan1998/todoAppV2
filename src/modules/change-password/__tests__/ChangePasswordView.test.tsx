import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import { ChangePasswordView } from '../ChangePasswordView';
import { useAuth } from 'src/contexts';

jest.mock('@next/third-parties/google', () => {
  return {
    GoogleAnalytics: () => <>GoogleAnalytics</>
  };
});

jest.mock('src/hooks', () => {
  const originalModule = jest.requireActual('src/hooks');
  return {
    ...originalModule,
    useMediaQuery: jest
      .fn()
      .mockReturnValue({
        isMobile: true
      })
      .mockReturnValueOnce({ isMobile: false })
      .mockReturnValueOnce({ isMobile: false })
  };
});

jest.mock('src/contexts', () => {
  return {
    useAuth: jest
      .fn()
      .mockReturnValue({ currentUser: { displayName: 'displayName', email: 'email' }, updatePassword: jest.fn() })
  };
});

jest.mock('next/navigation', () => {
  return {
    useRouter: jest.fn().mockReturnValue({
      push: jest.fn()
    })
  };
});

describe('ChangePasswordView', () => {
  it('renders ChangePasswordView with mobile view', () => {
    const component = render(<ChangePasswordView />);
    expect(component).toMatchSnapshot();
  });

  it('renders ChangePasswordView with desktop view', () => {
    const component = render(<ChangePasswordView />);
    const input2 = screen.getByLabelText('New password');
    fireEvent.change(input2, { target: { value: '123' } });
    const input3 = screen.getByLabelText('Confirm new password');
    fireEvent.change(input3, { target: { value: '123' } });

    const button = component.getByTestId('change-password-button');
    fireEvent.click(button);
    expect(component).toMatchSnapshot();
  });

  it('renders ChangePasswordView with desktop view', () => {
    const component = render(<ChangePasswordView />);
    const input2 = screen.getByLabelText('New password');
    fireEvent.change(input2, { target: { value: '123' } });
    const input3 = screen.getByLabelText('Confirm new password');
    fireEvent.change(input3, { target: { value: '12345' } });

    const button = component.getByTestId('change-password-button');
    fireEvent.click(button);
    expect(component).toMatchSnapshot();
  });

  it('renders ChangePasswordView with mobile view', () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    useAuth.mockImplementation(
      jest.fn(() => ({
        currentUser: { displayName: 'displayName', email: 'email' },
        updatePassword: jest.fn().mockRejectedValue('error')
      }))
    );
    const component = render(<ChangePasswordView />);
    const input2 = screen.getByLabelText('New password');
    fireEvent.change(input2, { target: { value: '123' } });
    const input3 = screen.getByLabelText('Confirm new password');
    fireEvent.change(input3, { target: { value: '123' } });

    const button = component.getByTestId('change-password-button');
    fireEvent.click(button);
    expect(component).toMatchSnapshot();
  });
});
