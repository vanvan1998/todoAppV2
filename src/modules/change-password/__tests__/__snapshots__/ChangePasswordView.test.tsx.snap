// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePasswordView renders ChangePasswordView with desktop view 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-UblHX ZlAPB"
      >
        <div
          class="sc-fufdwm eMJCuP"
        >
          <div
            class="sc-Rjrgp eCIyui"
          >
            <div
              class="sc-blHHSb kTKyqX"
              style="font-weight: 600;"
            >
              Change password
            </div>
            <div
              class="sc-egkSDF cGFidm"
            >
              Please enter your new password to change
            </div>
          </div>
          <div
            class="sc-gtLWhw jPWJOW"
          >
            You are login as
             
            <div
              class="sc-gtLWhw jPWJOW"
              style="font-weight: bold; white-space: pre-wrap; overflow-wrap: break-word;"
            >
              email
            </div>
          </div>
        </div>
        <div
          style="margin-bottom: 12px; margin-top: 40px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            New password
          </div>
          <input
            aria-label="New password"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your password..."
            style="min-width: 200px; border: 1px solid #858585; background-color: rgb(254, 254, 255); max-width: 320px; width: 80vw;"
            type="password"
            value="123"
          />
        </div>
        <div
          style="margin-bottom: 12px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            Confirm new password
          </div>
          <input
            aria-label="Confirm new password"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your password..."
            style="min-width: 200px; border: 1px solid #858585; background-color: rgb(254, 254, 255); max-width: 320px; width: 80vw;"
            type="password"
            value="123"
          />
        </div>
        <button
          buttontype="primary"
          class="sc-ghWlax hYhlzt"
          data-testid="change-password-button"
          disabled=""
          style="margin-top: 32px; margin-bottom: 16px; width: 80vw; max-width: 320px; min-width: 200px; background-color: rgb(182, 180, 185);"
          type="submit"
        >
          Change Password
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-UblHX ZlAPB"
    >
      <div
        class="sc-fufdwm eMJCuP"
      >
        <div
          class="sc-Rjrgp eCIyui"
        >
          <div
            class="sc-blHHSb kTKyqX"
            style="font-weight: 600;"
          >
            Change password
          </div>
          <div
            class="sc-egkSDF cGFidm"
          >
            Please enter your new password to change
          </div>
        </div>
        <div
          class="sc-gtLWhw jPWJOW"
        >
          You are login as
           
          <div
            class="sc-gtLWhw jPWJOW"
            style="font-weight: bold; white-space: pre-wrap; overflow-wrap: break-word;"
          >
            email
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 12px; margin-top: 40px;"
      >
        <div
          class="sc-gtLWhw jPWJOW"
        >
          New password
        </div>
        <input
          aria-label="New password"
          class="sc-kLhKbu gjDzjv"
          placeholder="Enter your password..."
          style="min-width: 200px; border: 1px solid #858585; background-color: rgb(254, 254, 255); max-width: 320px; width: 80vw;"
          type="password"
          value="123"
        />
      </div>
      <div
        style="margin-bottom: 12px;"
      >
        <div
          class="sc-gtLWhw jPWJOW"
        >
          Confirm new password
        </div>
        <input
          aria-label="Confirm new password"
          class="sc-kLhKbu gjDzjv"
          placeholder="Enter your password..."
          style="min-width: 200px; border: 1px solid #858585; background-color: rgb(254, 254, 255); max-width: 320px; width: 80vw;"
          type="password"
          value="123"
        />
      </div>
      <button
        buttontype="primary"
        class="sc-ghWlax hYhlzt"
        data-testid="change-password-button"
        disabled=""
        style="margin-top: 32px; margin-bottom: 16px; width: 80vw; max-width: 320px; min-width: 200px; background-color: rgb(182, 180, 185);"
        type="submit"
      >
        Change Password
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ChangePasswordView renders ChangePasswordView with desktop view 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-UblHX ZlAPB"
      >
        <div
          class="sc-fufdwm eMJCuP"
        >
          <div
            class="sc-Rjrgp eCIyui"
          >
            <div
              class="sc-blHHSb kTKyqX"
              style="font-weight: 600;"
            >
              Change password
            </div>
            <div
              class="sc-egkSDF cGFidm"
            >
              Please enter your new password to change
            </div>
          </div>
          <div
            class="sc-gtLWhw jPWJOW"
          >
            You are login as
             
            <div
              class="sc-gtLWhw jPWJOW"
              style="font-weight: bold; white-space: pre-wrap; overflow-wrap: break-word;"
            >
              email
            </div>
          </div>
        </div>
        <div
          style="margin-bottom: 12px; margin-top: 40px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            New password
          </div>
          <input
            aria-label="New password"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your password..."
            style="max-width: 320px; min-width: 200px; width: 80vw; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
            type="password"
            value="123"
          />
        </div>
        <div
          style="margin-bottom: 12px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            Confirm new password
          </div>
          <input
            aria-label="Confirm new password"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your password..."
            style="max-width: 320px; min-width: 200px; width: 80vw; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
            type="password"
            value="12345"
          />
        </div>
        <div
          class="sc-frniUE hsGpvU"
        >
          <div
            class="sc-dntaoT cZqBTg"
          >
            Passwords do not match
          </div>
        </div>
        <button
          buttontype="primary"
          class="sc-ghWlax hYhlzt"
          data-testid="change-password-button"
          style="margin-top: 32px; margin-bottom: 16px; max-width: 320px; min-width: 200px; width: 80vw;"
          type="submit"
        >
          Change Password
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-UblHX ZlAPB"
    >
      <div
        class="sc-fufdwm eMJCuP"
      >
        <div
          class="sc-Rjrgp eCIyui"
        >
          <div
            class="sc-blHHSb kTKyqX"
            style="font-weight: 600;"
          >
            Change password
          </div>
          <div
            class="sc-egkSDF cGFidm"
          >
            Please enter your new password to change
          </div>
        </div>
        <div
          class="sc-gtLWhw jPWJOW"
        >
          You are login as
           
          <div
            class="sc-gtLWhw jPWJOW"
            style="font-weight: bold; white-space: pre-wrap; overflow-wrap: break-word;"
          >
            email
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 12px; margin-top: 40px;"
      >
        <div
          class="sc-gtLWhw jPWJOW"
        >
          New password
        </div>
        <input
          aria-label="New password"
          class="sc-kLhKbu gjDzjv"
          placeholder="Enter your password..."
          style="max-width: 320px; min-width: 200px; width: 80vw; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
          type="password"
          value="123"
        />
      </div>
      <div
        style="margin-bottom: 12px;"
      >
        <div
          class="sc-gtLWhw jPWJOW"
        >
          Confirm new password
        </div>
        <input
          aria-label="Confirm new password"
          class="sc-kLhKbu gjDzjv"
          placeholder="Enter your password..."
          style="max-width: 320px; min-width: 200px; width: 80vw; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
          type="password"
          value="12345"
        />
      </div>
      <div
        class="sc-frniUE hsGpvU"
      >
        <div
          class="sc-dntaoT cZqBTg"
        >
          Passwords do not match
        </div>
      </div>
      <button
        buttontype="primary"
        class="sc-ghWlax hYhlzt"
        data-testid="change-password-button"
        style="margin-top: 32px; margin-bottom: 16px; max-width: 320px; min-width: 200px; width: 80vw;"
        type="submit"
      >
        Change Password
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ChangePasswordView renders ChangePasswordView with mobile view 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-UblHX ZlAPB"
      >
        <div
          class="sc-fufdwm eMJCuP"
        >
          <div
            class="sc-Rjrgp eCIyui"
          >
            <div
              class="sc-blHHSb kTKyqX"
              style="font-weight: 600;"
            >
              Change password
            </div>
            <div
              class="sc-egkSDF cGFidm"
            >
              Please enter your new password to change
            </div>
          </div>
          <div
            class="sc-gtLWhw jPWJOW"
          >
            You are login as
             
            <div
              class="sc-gtLWhw jPWJOW"
              style="font-weight: bold; white-space: pre-wrap; overflow-wrap: break-word;"
            >
              email
            </div>
          </div>
        </div>
        <div
          style="margin-bottom: 12px; margin-top: 40px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            New password
          </div>
          <input
            aria-label="New password"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your password..."
            style="min-height: 48px; min-width: 320px; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
            type="password"
            value=""
          />
        </div>
        <div
          style="margin-bottom: 12px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            Confirm new password
          </div>
          <input
            aria-label="Confirm new password"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your password..."
            style="min-height: 48px; min-width: 320px; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
            type="password"
            value=""
          />
        </div>
        <button
          buttontype="primary"
          class="sc-ghWlax hYhlzt"
          data-testid="change-password-button"
          disabled=""
          style="margin-top: 32px; margin-bottom: 16px; width: 320px; background-color: rgb(182, 180, 185);"
          type="submit"
        >
          Change Password
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-UblHX ZlAPB"
    >
      <div
        class="sc-fufdwm eMJCuP"
      >
        <div
          class="sc-Rjrgp eCIyui"
        >
          <div
            class="sc-blHHSb kTKyqX"
            style="font-weight: 600;"
          >
            Change password
          </div>
          <div
            class="sc-egkSDF cGFidm"
          >
            Please enter your new password to change
          </div>
        </div>
        <div
          class="sc-gtLWhw jPWJOW"
        >
          You are login as
           
          <div
            class="sc-gtLWhw jPWJOW"
            style="font-weight: bold; white-space: pre-wrap; overflow-wrap: break-word;"
          >
            email
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 12px; margin-top: 40px;"
      >
        <div
          class="sc-gtLWhw jPWJOW"
        >
          New password
        </div>
        <input
          aria-label="New password"
          class="sc-kLhKbu gjDzjv"
          placeholder="Enter your password..."
          style="min-height: 48px; min-width: 320px; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
          type="password"
          value=""
        />
      </div>
      <div
        style="margin-bottom: 12px;"
      >
        <div
          class="sc-gtLWhw jPWJOW"
        >
          Confirm new password
        </div>
        <input
          aria-label="Confirm new password"
          class="sc-kLhKbu gjDzjv"
          placeholder="Enter your password..."
          style="min-height: 48px; min-width: 320px; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
          type="password"
          value=""
        />
      </div>
      <button
        buttontype="primary"
        class="sc-ghWlax hYhlzt"
        data-testid="change-password-button"
        disabled=""
        style="margin-top: 32px; margin-bottom: 16px; width: 320px; background-color: rgb(182, 180, 185);"
        type="submit"
      >
        Change Password
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ChangePasswordView renders ChangePasswordView with mobile view 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-UblHX ZlAPB"
      >
        <div
          class="sc-fufdwm eMJCuP"
        >
          <div
            class="sc-Rjrgp eCIyui"
          >
            <div
              class="sc-blHHSb kTKyqX"
              style="font-weight: 600;"
            >
              Change password
            </div>
            <div
              class="sc-egkSDF cGFidm"
            >
              Please enter your new password to change
            </div>
          </div>
          <div
            class="sc-gtLWhw jPWJOW"
          >
            You are login as
             
            <div
              class="sc-gtLWhw jPWJOW"
              style="font-weight: bold; white-space: pre-wrap; overflow-wrap: break-word;"
            >
              email
            </div>
          </div>
        </div>
        <div
          style="margin-bottom: 12px; margin-top: 40px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            New password
          </div>
          <input
            aria-label="New password"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your password..."
            style="max-width: 320px; min-width: 200px; width: 80vw; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
            type="password"
            value="123"
          />
        </div>
        <div
          style="margin-bottom: 12px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            Confirm new password
          </div>
          <input
            aria-label="Confirm new password"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your password..."
            style="max-width: 320px; min-width: 200px; width: 80vw; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
            type="password"
            value="123"
          />
        </div>
        <button
          buttontype="primary"
          class="sc-ghWlax hYhlzt"
          data-testid="change-password-button"
          disabled=""
          style="margin-top: 32px; margin-bottom: 16px; max-width: 320px; min-width: 200px; width: 80vw; background-color: rgb(182, 180, 185);"
          type="submit"
        >
          Change Password
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-UblHX ZlAPB"
    >
      <div
        class="sc-fufdwm eMJCuP"
      >
        <div
          class="sc-Rjrgp eCIyui"
        >
          <div
            class="sc-blHHSb kTKyqX"
            style="font-weight: 600;"
          >
            Change password
          </div>
          <div
            class="sc-egkSDF cGFidm"
          >
            Please enter your new password to change
          </div>
        </div>
        <div
          class="sc-gtLWhw jPWJOW"
        >
          You are login as
           
          <div
            class="sc-gtLWhw jPWJOW"
            style="font-weight: bold; white-space: pre-wrap; overflow-wrap: break-word;"
          >
            email
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 12px; margin-top: 40px;"
      >
        <div
          class="sc-gtLWhw jPWJOW"
        >
          New password
        </div>
        <input
          aria-label="New password"
          class="sc-kLhKbu gjDzjv"
          placeholder="Enter your password..."
          style="max-width: 320px; min-width: 200px; width: 80vw; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
          type="password"
          value="123"
        />
      </div>
      <div
        style="margin-bottom: 12px;"
      >
        <div
          class="sc-gtLWhw jPWJOW"
        >
          Confirm new password
        </div>
        <input
          aria-label="Confirm new password"
          class="sc-kLhKbu gjDzjv"
          placeholder="Enter your password..."
          style="max-width: 320px; min-width: 200px; width: 80vw; border: 1px solid #858585; background-color: rgb(254, 254, 255);"
          type="password"
          value="123"
        />
      </div>
      <button
        buttontype="primary"
        class="sc-ghWlax hYhlzt"
        data-testid="change-password-button"
        disabled=""
        style="margin-top: 32px; margin-bottom: 16px; max-width: 320px; min-width: 200px; width: 80vw; background-color: rgb(182, 180, 185);"
        type="submit"
      >
        Change Password
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
